osu.Game.Rulesets.Taiko/Difficulty/Preprocessing/Rhythm/Data/SamePatterns.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace osu.Game.Rulesets.Taiko.Difficulty.Preprocessing.Rhythm.Data
+{
+    /// <summary>
+    /// Represents <see cref="SameRhythmHitObjects"/> grouped by their <see cref="SameRhythmHitObjects.StartTime"/>'s interval.
+    /// </summary>
+    public class SamePatterns : SameRhythm<SameRhythmHitObjects>
```

Comment:
this is oop hell. without reading any implementation, none of these class names sound sane.

Reply:
Originally, this was labelled as Even instead of Same, with both referring to the objects being grouped by their ratio (rhythm), thus being a same rhythm object, or a same rhythm pattern.